app/javascript/article-form/components/ArticleCoverImage.jsx

```diff
@@ -53,55 +58,75 @@ export class ArticleCoverImage extends Component {
     });
   };
 
+  onDropImage = (event) => {
+    onDragExit(event);
+
+    if (event.dataTransfer.files.length > 1) {
+      addSnackbarItem({
+        message: 'Only one image can be dropped at a time.',
+        addCloseButton: true,
+      });
+      return;
+    }
+
+    this.handleMainImageUpload(event);
+  };
+
   render() {
     const { mainImage } = this.props;
     const { uploadError, uploadErrorMessage, uploadingImage } = this.state;
     const uploadLabel = mainImage ? 'Change' : 'Add a cover image';
 
     return (
-      <div className="crayons-article-form__cover" role="presentation">
-        {!uploadingImage && mainImage && (
-          <img
-            src={mainImage}
-            className="crayons-article-form__cover__image"
-            width="250"
-            height="105"
-            alt="Post cover"
-          />
-        )}
-        <div className="flex items-center">
-          {uploadingImage ? (
-            <span class="lh-base pl-1 border-0 py-2 inline-block">
-              <Spinner /> Uploading...
-            </span>
-          ) : (
-            <Fragment>
-              <Button variant="outlined" className="mr-2 whitespace-nowrap">
-                <label htmlFor="cover-image-input">{uploadLabel}</label>
-                <input
-                  id="cover-image-input"
-                  type="file"
-                  onChange={this.handleMainImageUpload}
-                  accept="image/*"
-                  className="w-100 h-100 absolute left-0 right-0 top-0 bottom-0 overflow-hidden opacity-0 cursor-pointer"
-                  data-max-file-size-mb="25"
-                />
-              </Button>
-              {mainImage && (
-                <Button
-                  variant="ghost-danger"
-                  onClick={this.triggerMainImageRemoval}
-                >
-                  Remove
+      <DragAndDropZone
+        onDragOver={onDragOver}
+        onDragExit={onDragExit}
+        onDrop={this.onDropImage}
+      >
+        <div className="crayons-article-form__cover" role="presentation">
+          {!uploadingImage && mainImage && (
+            <img
+              src={mainImage}
+              className="crayons-article-form__cover__image"
+              width="250"
+              height="105"
+              alt="Post cover"
+            />
+          )}
+          <div className="flex items-center">
+            {uploadingImage ? (
+              <span class="lh-base pl-1 border-0 py-2 inline-block">
+                <Spinner /> Uploading...
```

Comment:
It'd be cool if we could see something like this for the article images as well but I read that a UX refactoring PR is in the pipeline

Reply:
Yeah exactly. For the cover image I basically follow the same flow as an image upload via the native file selector. For the post content it is a little different. @ludwiczakpawel is addressing it in #10203.